core/src/main/java/org/mobilitydata/gtfsvalidator/input/GtfsInput.java

```diff
@@ -44,6 +56,62 @@ static GtfsInput createFromPath(String path) throws IOException {
         return new GtfsZipFileInput(file);
     }
 
+    /**
+     * Creates an specific GtfsInput to read data from the given URL.
+     *
+     * @param sourceUrl           the fully qualified URL to download of the resource to download
+     * @param targetPathAsString  the path to where the downloaded resource will be stored
+     * @return the @code{GtfsInput} created after download of the GTFS archive
+     * @throws IOException  if no file could not be found at the specified location
+     * @throws URISyntaxException  if URL is malformed
+     * @throws InterruptedException when a thread is waiting, sleeping, or otherwise occupied, and the thread is
+     * interrupted, either before or during the activity
+     */
+    static GtfsInput createFromUrl(URL sourceUrl, String targetPathAsString) throws IOException, URISyntaxException,
+            InterruptedException {
+        CloseableHttpClient httpClient = HttpClients.createDefault();
+        HttpGet httpGet = new HttpGet(sourceUrl.toURI());
+        CloseableHttpResponse httpResponse = httpClient.execute(httpGet);
+        InputStream inputStream = httpResponse.getEntity().getContent();
```

Comment:
Looking at https://hc.apache.org/httpcomponents-client-ga/tutorial/pdf/httpclient-tutorial.pdf, in section 1.1.5 it says:

>In order to ensure proper release of system resources one must close either the content stream associated
with the entity or the response itself
>...
>The difference between closing the content stream and closing the response is that the former will
attempt to keep the underlying connection alive by consuming the entity content while the latter
immediately shuts down and discards the connection.
>
>Please note that the HttpEntity#writeTo(OutputStream) method is also required to ensure proper
release of system resources once the entity has been fully written out. If this method obtains an instance
of java.io.InputStream by calling HttpEntity#getContent(), it is also expected to close the stream
in a finally clause.

It seems safer to use the `HttpEntity.writeTo()` method instead of `getContent()` as the `OutputStream` is managed by the `HttpEntity` (also please be sure to use a BufferedOutputStream). I believe that would also allow us to avoid `try/catch/finally` block within this method if you want to throw the exception to the calling code.